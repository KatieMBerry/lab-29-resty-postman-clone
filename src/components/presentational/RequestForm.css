.RequestForm{
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  header{
      background-color: black;
      color: white;
      padding: 20px;
      width: 100vw;
  }
  
  section{
    display: flex;
    width: 100%;
    justify-content: flex-start;
    align-items: center;
    flex-direction: row;

    input[type="radio"]{
      visibility: hidden;
      height: 0;
      width: 0;
      margin-left: 0;
    }
  
    label{
      padding: 4px 16px;
      cursor: pointer;
    }

    input[type="radio"]:checked+label{
      background-color: #204060;
      color: white;
    }
  }

  input{
    margin-bottom: 8px;
  }

  button{
    margin-left: 2vw;
  }

  textarea{
    margin-top: 5px;
    width: 50%;
    height: 6vw;
    align-self: flex-start;
  }
}

.url{
  width: 200px;
  height: 25px;
  font-size: 12px;
  margin-top: 5px;
}
