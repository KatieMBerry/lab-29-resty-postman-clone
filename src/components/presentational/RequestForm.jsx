import React from 'react';
import PropTypes from 'prop-types';

const RequestForm = ({ urlInput, selectedMethod, requestBody, onSubmit, onChange }) => (

    <section>
        <form onSubmit={onSubmit} >
            <input type="text"
                placeholder="url"
                name="url"
                value={urlInput}
                onChange={onChange} />

            <div>
                <label htmlFor="GET">GET</label>
                <input type="radio"
                    id="GET"
                    name="method"
                    value="GET"
                    onChange={onChange}
                    checked={selectedMethod === 'GET'} />

                <label htmlFor="POST">POST</label>
                <input type="radio"
                    id="GET"
                    name="method"
                    value="POST"
                    onChange={onChange}
                    checked={selectedMethod === 'POST'} />

                <label htmlFor="PUT">PUT</label>
                <input type="radio"
                    id="PUT"
                    name="method"
                    value="PUT"
                    onChange={onChange}
                    checked={selectedMethod === 'PUT'} />

                <label htmlFor="DELETE">DELETE</label>
                <input type="radio"
                    id="DELETE"
                    name="method"
                    value="DELETE"
                    onChange={onChange}
                    checked={selectedMethod === 'DELETE'} />
            </div>
            <button>Fetch API</button>

            <textarea placeholder="Raw JSON"
                name="request body"
                value={requestBody}
                onChange={onChange}></textarea>
        </form>
    </section>
);

RequestForm.propTypes = {
    urlInput: PropTypes.string.isRequired,
    selectedMethod: PropTypes.string.isRequired,
    requestBody: PropTypes.string.isRequired,
    onSubmit: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired
}

export default RequestForm;
