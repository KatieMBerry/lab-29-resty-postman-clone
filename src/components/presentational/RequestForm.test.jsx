import React from 'react';
import { render, cleanup } from '@testing-library/react';
import RequestForm from './RequestForm';

describe('RequestForm component', () => {
    afterEach(() => cleanup());
    it('renders RequestForm', () => {
        const { asFragment } = render(<RequestForm
            onSubmit={jest.fn()}
            urlInput='my_url'
            onChange={jest.fn()}

            selectedMethod='GET'
            onChange={jest.fn()}

            requestBody='my_json'
            onChange={jest.fn()}
        />);

        expect(asFragment()).toMatchSnapshot();
    })
})
