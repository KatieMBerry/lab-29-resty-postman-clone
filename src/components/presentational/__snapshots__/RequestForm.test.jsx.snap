// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestForm component renders RequestForm 1`] = `
<DocumentFragment>
  <section>
    <form>
      <input
        name="url"
        placeholder="url"
        type="text"
        value="my_url"
      />
      <div>
        <label
          for="GET"
        >
          GET
        </label>
        <input
          checked=""
          id="GET"
          name="method"
          type="radio"
          value="GET"
        />
        <label
          for="POST"
        >
          POST
        </label>
        <input
          id="GET"
          name="method"
          type="radio"
          value="POST"
        />
        <label
          for="PUT"
        >
          PUT
        </label>
        <input
          id="PUT"
          name="method"
          type="radio"
          value="PUT"
        />
        <label
          for="DELETE"
        >
          DELETE
        </label>
        <input
          id="DELETE"
          name="method"
          type="radio"
          value="DELETE"
        />
      </div>
      <button>
        Fetch API
      </button>
      <textarea
        name="request body"
        placeholder="Raw JSON"
      >
        my_json
      </textarea>
    </form>
  </section>
</DocumentFragment>
`;
